Etabs_Adapter/CRUD/Read/Results/GlobalResults.cs

```diff
@@ -135,6 +139,54 @@ private List<ModalDynamics> GetModalParticipationMassRatios()
 
         /***************************************************/
 
+        private List<StoreyDrift> GetStoreyDrifts()
+        {
+            List<StoreyDrift> storeyDrifts = new List<StoreyDrift>();
+
+            int resultCount = 0;
+            string[] loadcaseNames = null;
+            string[] stepType = null; double[] stepNum = null;
+            string[] storeyNames = null; string[] directions = null;
+            double[] drifts = null; string[] labels = null;
+            double[] x = null; double[] y= null; double[] z = null;
+
+            m_model.Results.StoryDrifts(ref resultCount, ref storeyNames, ref loadcaseNames, ref stepType, ref stepNum, ref directions, ref drifts, ref labels, ref x, ref y, ref z);
+
+            Vector[] dirVectors = directions.Select(dirStr =>
+                {
+                    switch (dirStr.ToUpper())
+                    {
+                        case "X":
+                            return Vector.XAxis;
+                            break;
+                        case "Y":
+                            return Vector.YAxis;
+                            break;
+                        default:
+                            return Vector.ZAxis;
+                            break;
+                    }
+                }).ToArray();
```

Comment:
@peterjamesnugent, thanks for the comment.
I think that this can stay. The directions output from ETABS via its API (in the code line above the snippet), are string objects with value "X"/"Y"/"Z".
The switch case block above is converting the string data into BHoM Vector data to be assigned to the BHoM StoreyDrift object.
Let me know in case I'm missing sth ;)